rust/main/chains/hyperlane-ethereum/src/rpc_clients/fallback.rs

```diff
@@ -90,8 +94,65 @@ where
     where
         T: Debug + Serialize + Send + Sync,
         R: DeserializeOwned,
+    {
+        if method == METHOD_SEND_RAW_TRANSACTION {
+            self.multicast(method, params).await
+        } else {
+            self.fallback(method, params).await
+        }
+    }
+}
+
+impl<C> EthereumFallbackProvider<C, JsonRpcBlockGetter<C>>
+where
+    C: JsonRpcClient<Error = HttpClientError>
+        + Into<JsonRpcBlockGetter<C>>
+        + PrometheusJsonRpcClientConfigExt
+        + Clone,
+    JsonRpcBlockGetter<C>: BlockNumberGetter,
+{
+    async fn multicast<T, R>(&self, method: &str, params: T) -> Result<R, ProviderError>
+    where
+        T: Serialize,
+        R: DeserializeOwned,
     {
         use CategorizedResponse::*;
+
+        let params = serde_json::to_value(params).expect("valid");
+
+        // errors reported by providers
+        let mut errors = vec![];
+
+        // retry 4 times if all providers returned a retryable error
+        for i in 0..=3 {
+            if i > 0 {
+                // sleep starting from the second attempt
+                sleep(Duration::from_millis(100)).await;
+            }
+
+            // future which visits all providers as they fulfill their requests
+            let mut unordered = self.populate_unordered_future(method, &params);
+
+            while let Some(resp) = unordered.next().await {
+                match categorize_client_response(method, resp) {
```

Comment:
curious if you think we should do a `_span` somewhere in here like we do for fallback, or if we already have all we can possibly need for the logs

Reply:
I added a warn span in the case when we did not receive a successful result from all the providers. It means that we shall see the warn span at most 4 times, printed at every attempts of the outer `for` loop.